src/environment.js

```diff
@@ -1151,6 +1176,9 @@ export class LexicalEnvironment {
       res = this.evaluateCompletion(ast, false);
     } finally {
       this.realm.popContext(context);
+      if (context.lexicalEnvironment !== this) this.realm.onDestroyScope(context.lexicalEnvironment);
```

Comment:
Will this always restore context.lexicalEnvironment to this? If so, it would be nice to say so in an invariant.

Reply:
It doesn't restore to `this`. `onDestroyScope` no longer restores anything. The if is necessary because in exceptional cases, the lexicalEnvironment is the environment that threw I believe.